### ISSUE TITLE
Autoscroll implementation for TreeDataGrid

### ISSUE THREAD

**Issue by Dave-Krch** [AvaloniaUI/Avalonia.Controls.TreeDataGrid#324]
I have managed to scroll TreeDataGrid using this method in MainWindow.axaml.cs:
```
private void ScrollTreeDataGrid()
{
    var vm = DataContext as MainWindowViewModel;
    if (vm != null && vm.AutoScrollData)
        tdGrid.Scroll.Offset = tdGrid.Scroll.Offset.WithY(tdGrid.Scroll.Extent.Height);
}
```
I need to call this every time the source for TreeDataGrid is updated, but i did not find any event that i can use.
Items are being added to the Source using this method in ManWindowVIewModel:
```
public void AddItem(ASN1Type msg)
{
    DecodedMessages.Add(new ReceivedMessageModel(MsgCount, DateTime.Now, v2Xinterface.Config.Port, v2Xinterface.Config.IPAddress, msg));
    MsgCount++;
}
```
Is there an event form TreeDataGrid that i am mising? Or is there a way i can acces the TreeDataGrid from the ViewModel and set the scroll there?

**Comment by Dave-Krch** [comment]
In case anyone has this problem in future, this is how i solved it:
TLDR: I created method that scrolls to end in code behind and call it every 100ms in MainWindowViewModel

MainWindow.axaml:

```
<TreeDataGrid Name="tdGrid" Source="{Binding Source}"/>
```

MainWindow.axaml.cs:
```
private MainWindowViewModel _mwvm;

public MainWindow()
{
    InitializeComponent();
    _mwvm = new MainWindowViewModel();
    DataContext = _mwvm;

    _mwvm.ScrollAction = () => ScrollMessages();
}

private void ScrollMessages()
{
    tdGrid.Scroll.Offset = new Avalonia.Vector(tdGrid.Scroll.Offset.X, tdGrid.Scroll.Extent.Height - tdGrid.Scroll.Viewport.Height);
}

```

MainWindowViewModel.cs
```
public Action? ScrollAction { get; set; }

...

private void TimerEvent(Object? source, ElapsedEventArgs e)
{

    if (AutoScrollData)
    {
        if(ScrollAction == null)
        {
            Dispatcher.UIThread.InvokeAsync(() => ConsoleMessages.Add("ScrollAction not set"));
        }
        else
        {
            Dispatcher.UIThread.InvokeAsync(() => ScrollAction());
        }
    }

}

```
